homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,204 @@
+                        self.session.data.minmax[nodeid]['TodayMax'] = \
+                            curtempret
+
+                    if (curtempret < self.session.data.minmax[nodeid]
+                            ['RestartMin']):
+                        self.session.data.minmax[nodeid]['RestartMin'] = \
+                            curtempret
+
+                    if (curtempret >
+                            self.session.data.minmax[nodeid]
+                            ['RestartMax']):
+                        self.session.data.minmax[nodeid]['RestartMax'] = \
+                            curtempret
+                else:
+                    current_node_max_min_data = {}
+                    current_node_max_min_data['TodayMin'] = curtempret
+                    current_node_max_min_data['TodayMax'] = curtempret
+                    current_node_max_min_data['TodayDate'] = \
+                        datetime.date(datetime.now())
+                    current_node_max_min_data['RestartMin'] = curtempret
+                    current_node_max_min_data['RestartMax'] = curtempret
+                    self.session.data.minmax[nodeid] = \
+                        current_node_max_min_data
+            else:
+                curtempret = 0
+            return curtempret
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        set_result = None
+
+        if self.device_type == "Heating":
+            set_result = self.session.heating.get_target_temperature(
+                self.node_id)
+        elif self.device_type == "HotWater":
+            set_result = None
+
+        return set_result
+
+    @property
+    def min_temp(self):
+        """Return minimum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.min_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.device_type == "Heating":
+            return self.session.heating.max_temperature(self.node_id)
+        elif self.device_type == "HotWater":
+            return None
+
+    @property
+    def operation_list(self):
+        """List of the operation modes."""
+        if self.device_type == "Heating":
+            return self.session.heating.get_operation_modes(self.node_id)
+        elif self.device_type == "HotWater":
+            return self.session.hotwater.get_operation_modes(self.node_id)
+
+    @property
+    def current_operation(self):
+        """Return current mode."""
+        if self.device_type == "Heating":
+            return self.session.heating.get_mode(self.node_id)
+        elif self.device_type == "HotWater":
+            return self.session.hotwater.get_mode(self.node_id)
+
+    def set_operation_mode(self, operation_mode):

```

Comment:
See `operation_list()`

Reply:
as ```operation_list()```

Reply:
as above.
I have updated to use the home assistant standards